lib/govuk_ab_testing/abstract_helpers.rb

```diff
@@ -0,0 +1,89 @@
+module GovukAbTesting
+  module AbstractHelpers
+    def acceptance_test_framework
+      @acceptance_test_framework ||= GovukAbTesting.configuration.framework_class.new(self)
+    end
+
+    def with_variant(args)
+      ab_test_name, variant = args.first
+      # AKG TODO: Should this be required?
+      dimension = args[:dimension]
+
+      setup_ab_variant(ab_test_name, variant, dimension)
+
+      yield
+
+      assert_response_is_cached_by_variant(ab_test_name)
+
+      unless args[:assert_meta_tag] == false
+        assert_page_tracked_in_ab_test(ab_test_name, variant, dimension)
+      end
+    end
+
+    def setup_ab_variant(ab_test_name, variant, dimension = 300)
+      ab_test = AbTest.new(ab_test_name, dimension: dimension)
+      acceptance_test_framework.set_header(ab_test.request_header, variant)
+    end
+
+    def assert_response_is_cached_by_variant(ab_test_name)
+      ab_test = AbTest.new(ab_test_name, dimension: 300)
+      vary_header_value = acceptance_test_framework.vary_header
+
+      assert_contains_substring(
+        string: vary_header_value,
+        substring: ab_test.response_header,
+        error_message: "'Vary' header is not being set for the A/B test '#{ab_test.name}'"
+      )
+    end
+
+    def assert_response_not_modified_for_ab_test(ab_test_name)
+      vary_header = acceptance_test_framework.vary_header
+      assert_does_not_contain_substring(
+        string: vary_header,
+        substring: ab_test_name,
+        error_message: "'Vary' header is being added to a page which should not be modified by the A/B test '#{ab_test_name}'",
+      )
+    end
+
+    def assert_page_not_tracked_in_ab_test(ab_test_name)
+      ab_test_meta_tags =
+        acceptance_test_framework.analytics_meta_tags_for_test(ab_test_name)
+
+      assert_is_empty(
+        enumerable: ab_test_meta_tags,
+        error_message: "A/B meta tag is being added to a page which should not be modified by the A/B test"
+      )
+    end
+
+    def assert_page_tracked_in_ab_test(ab_test_name, variant, dimension)
+      ab_test = AbTest.new(ab_test_name, dimension: dimension)
+
+      ab_test_meta_tags =
+        acceptance_test_framework.analytics_meta_tags_for_test(ab_test.name)
+
+      assert_has_size(
+        enumerable: ab_test_meta_tags,
+        size: 1,
+        error_message: "'Found multiple analytics meta tags on the page for A/B test '#{ab_test.name}', but expected only 1"
+      )
+
+      meta_tag = ab_test_meta_tags.first
+      expected_metatag_content = "#{ab_test.meta_tag_name}:#{variant}"
+
+      assert_is_equal(
+        expected: expected_metatag_content,
+        actual: meta_tag.content,
+        error_message: "The analytics meta tag is not present on the page for A/B test '#{ab_test.name}'"
+      )
+
+      # AKG TODO: Should this be required?
+      unless ab_test.dimension.nil?
```

Comment:
Should we require `dimension` to not be `nil`? This would be a breaking change (in that it would break existing tests that are currently passing). In general, we've not been particularly consistent with how required `dimension` is, demanding it eg in the constructor for `AbTest`, but allowing dummy value defaults in these test helpers.